### ISSUE TITLE
How to modify the OminiControl if I have 2 imgs condition input C_I_1 and C_I_2?

### ISSUE THREAD

**Issue by BugsMaker0513** [Yuanshi9815/OminiControl#94]

**Comment by Yuanshi9815** [comment]
Hi @BugsMaker0513 ,

So far, the released models do not support multi-conditioning yet.
However, it is a part of our ongoing work.

:D

**Comment by sriharigolla** [comment]
Hi Yuanshi9815,

I appreciate your excellent work on OminiControl! I'm currently exploring multi-condition support, but I'm struggling to extend it to multiple subjects.

Could you please provide guidance or point me toward resources or code examples on implementing multi-condition support effectively? I'm eager to contribute and would greatly appreciate your assistance.

**Comment by Junsheng121** [comment]
Hi @Yuanshi9815 !

Appreciate for your impressive work! I'm also exploring the multi-subject generation based on OminiControl. I have tried to input two different subject images, but they tend to mix with each other. Now I'm working on this.

I notice the interesting dataset construction, subject200k, which use FLUX to generate the same object in a side-by-side image pair. I'd like to learn about your experience—have you considered using FLUX in a similar way to create a multi-subject dataset?

**Comment by sriharigolla** [comment]
Hello @Junsheng121 
I’d love to learn more about your approach and any insights you’ve gained so far. Have you made any progress in addressing the issue? It would be great to exchange ideas so we can refine our approaches and avoid repeating the same challenges.

**Comment by jschoormans** [comment]
Hi @Junsheng121 and @sriharigolla , I'm exploring the same things. I too found that adding multiple subject images mixed tried (I tried concatenating the latents and playing with the condition ids). Would love to discuss this with you too

**Comment by Junsheng121** [comment]
Hi @sriharigolla and @jschoormans!
In Subject200k I find that most of the [subject] is in the middle of an image. After the training, I guess the multiple subjects will also appear in the middle of the generated image, then blending. I believe a routing module is beneficial for multi-subject generation so that they can be generated in the correct region.

**Comment by Junsheng121** [comment]
> Interesting - I wonder if that can be achieved with attention masks? Don't really know how to approach that

This paper may help: Anystory https://arxiv.org/abs/2501.09503

**Comment by ewwwgiddings** [comment]
> Hi [@BugsMaker0513](https://github.com/BugsMaker0513) ,
> 
> So far, the released models do not support multi-conditioning yet. However, it is a part of our ongoing work.
> 
> :D

Is this on a roadmap or in a branch I can follow?

Would love to be able to combine the subject and fill conditions to guide the fill based on the subject.

Thanks for all the effort you put in!

**Comment by KleinXin** [comment]
> Hi [@BugsMaker0513](https://github.com/BugsMaker0513) ,
> 
> So far, the released models do not support multi-conditioning yet. However, it is a part of our ongoing work.
> 
> :D

Is the example 2 in Figure 8 （penguin example）of the paper a use case of multi-conditions？
The penguin image is the subject condition and the table image is the fill condition.
The codes of generate can support multi-conditions now, but I have not tried.
I will have a try since I can not find any multi-condition examples.
